### PR TITLE
docs: Update index-related config for 8.0

### PR DESCRIPTION
**Pull request by bmorelli25**
### Summary

This PR updates indexing-related configuration docs for standalone (legacy) mode. Specific removals and updates are outlined below.

**A preview of this PR is available [here](https://apm-server_6940.docs-preview.app.elstc.co/guide/en/apm/guide/master/index.html).**

---

**Index management configuration**

> Fleet will now be entirely responsible for setting up index templates, ILM policies, and index pipelines. Thus, all index management configuration is now removed from apm-server

- [x] `apm-server.ilm.*` docs
  - [x] Remove [Configure ILM](https://www.elastic.co/guide/en/apm/guide/8.0/ilm-reference.html)
  - [x] Update [Custom ILM](https://www.elastic.co/guide/en/apm/guide/8.0/ilm.html)
- [x] `apm-server.register.ingest.pipeline.*` docs
  - [x] Remove config docs
  ~- [ ] Update [Parse data using ingest node pipelines](https://www.elastic.co/guide/en/apm/guide/8.0/configuring-ingest-node.html)~ Removed as changes don't persist when upgraded
- [x] `setup.*` docs:
  - [x] Remove [Configure ES indes template loading](https://www.elastic.co/guide/en/apm/guide/8.0/configuration-template.html)
  ~- [ ] Update [Load the ES index template](https://www.elastic.co/guide/en/apm/guide/8.0/apm-server-template.html)~ Removed as changes don't persist when upgraded
- [x] Remove `setup` command docs

**Data streams**

> APM Server will only write to well-defined data streams, and will rely on the ingest pipeline being defined in index templates.

- [x] Remove `apm-server.data_streams.enabled` docs
- [x] Remove `output.elasticsearch.index` docs
- [x] Remove `output.elasticsearch.indices` docs
- [x] Remove `output.elasticsearch.pipeline` docs
- [x] Remove `output.elasticsearch.pipelines` docs
- [x] Remove [`resource \'apm-{v}-$type' exists, but it is not an alias`](https://www.elastic.co/guide/en/apm/guide/8.0/common-problems.html#server-resource-exists-not-alias) docs as data streams fixes the problem

**Elasticsearch output**

> APM Server has a new Elasticsearch output implementation with different configuration parameters.

- [x] Remove `output.elasticsearch.worker` docs
- [x] Remove `output.elasticsearch.bulk_max_size` docs
- [x] Remove `queue.*` docs (ES output only)
- [x] Add `output.elasticsearch.flush_bytes` docs
- [x] Add `output.elasticsearch.flush_interval` docs
- [x] Remove [Tune APM Server output parameters for your Elasticsearch cluster](https://www.elastic.co/guide/en/apm/guide/8.0/tune-apm-server.html#tune-output-config) as defaults are sufficient
- [x] Remove [Adjust internal queue size](https://www.elastic.co/guide/en/apm/guide/8.0/tune-apm-server.html#adjust-queue-size) as there is no internal queue to tune

**Roles and privileges**

- [x] Remove [Grant privileges and roles needed for setup](https://www.elastic.co/guide/en/apm/guide/8.0/privileges-to-setup-beats.html)
- [x] Update [Grant privileges and roles needed for writing events](https://www.elastic.co/guide/en/apm/guide/8.0/privileges-to-publish-events.html)

### Related issues

Closes https://github.com/elastic/apm-server/issues/6662.

### CODE REVIEW

**Comment by bmorelli25** [comment]
I'm still waiting for a review on https://github.com/elastic/kibana/pull/122397#issuecomment-1010280622 to unblock the doc build check. In the meantime, I have a few questions...

File: docs/ingest-pipelines.asciidoc
Comment (bmorelli25) [inline docs/ingest-pipelines.asciidoc:24]: Default ingest pipelines: should we doc which pipelines apply each stream?

File: docs/legacy/feature-roles.asciidoc
Comment (bmorelli25) [inline docs/legacy/feature-roles.asciidoc:35]: How are privileges for writing events impacted by data stream changes? Are [these docs](https://www.elastic.co/guide/en/apm/server/current/privileges-to-publish-events.html) still accurate?

File: docs/ingest-pipelines.asciidoc
Comment (bmorelli25) [inline docs/ingest-pipelines.asciidoc:90]: Before I put time into creating this custom ingest pipeline tutorial... does this rough outline sound correct? I tested this flow and everything _seems_ to work.

File: docs/custom-index-template.asciidoc
Comment (bmorelli25) [inline docs/custom-index-template.asciidoc:31]: Before I put time into creating this custom index/component template tutorial... does this rough outline sound correct? I tested this flow and everything _seems_ to work.

File: docs/legacy/common-problems.asciidoc
Comment (bmorelli25) [inline docs/legacy/common-problems.asciidoc:260]: Since the queue is gone in 8.0+, does that mean that if ES goes down, all data is lost?

File: docs/legacy/getting-started-apm-server.asciidoc
Comment (bmorelli25) [inline docs/legacy/getting-started-apm-server.asciidoc:257]: Should we doc a workaround like this for installing the APM integration?

File: docs/legacy/common-problems.asciidoc
Comment (axw) [reply docs/legacy/common-problems.asciidoc:260]: The queue isn't gone, it's just different :)

There's still an in-memory queue. If it fills up, the server will block until there is space, or the request timeout is reached.

The new queue shouldn't need tuning. I'd suggest dropping the "adjust the internal queue size" link and section.

File: docs/ingest-pipelines.asciidoc
Comment (axw) [reply docs/ingest-pipelines.asciidoc:90]: Sounds right, but I'm wondering if we should be documenting this. Not being able to edit pipelines in a way that persists is a known issue, but we don't know how much of a pain that is yet. By documenting the workaround so prominently we may not learn that. WDYT @simitt?

File: docs/legacy/getting-started-apm-server.asciidoc
Comment (axw) [reply docs/legacy/getting-started-apm-server.asciidoc:257]: Not sure I understand what the workaround is for. Why aren't the manual steps above enough?

File: docs/ingest-pipelines.asciidoc
Comment (bmorelli25) [reply docs/ingest-pipelines.asciidoc]: That's a good point—they are an implementation detail. Removed in 21892a6. We can always add them back in a later PR if Silvia feels strongly otherwise.

File: docs/legacy/feature-roles.asciidoc
Comment (bmorelli25) [reply docs/legacy/feature-roles.asciidoc:35]: Thanks! Fixed in 7eaea87.

File: docs/ingest-pipelines.asciidoc
Comment (bmorelli25) [reply docs/ingest-pipelines.asciidoc]: Thanks. Removing from this PR for now (51b4dca). Added your thoughts to the pipeline doc issue: https://github.com/elastic/observability-docs/issues/1226

File: docs/legacy/common-problems.asciidoc
Comment (bmorelli25) [reply docs/legacy/common-problems.asciidoc]: Ahhh. Thanks! Fixed in 780f7a3.

File: docs/custom-index-template.asciidoc
Comment (bmorelli25) [reply docs/custom-index-template.asciidoc]: For now, I've removed custom index template instructions to match what we have for custom pipelines.

**Comment by bmorelli25** [comment]
Merged https://github.com/elastic/kibana/pull/123072 to work around https://github.com/elastic/kibana/pull/122397/. That _should_ unblock the doc build for this PR 🤞.

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

File: docs/legacy/getting-started-apm-server.asciidoc
Comment (simitt) [inline docs/legacy/getting-started-apm-server.asciidoc:239]: This needs to be mentioned somewhere prominent. If users do not follow this, the APM Server generally starts up but eventually fills up the internal queue. Until then the issue is only observable through logs and missing data in the APM UI.

File: docs/ingest-pipelines.asciidoc
Comment (simitt) [reply docs/ingest-pipelines.asciidoc]: SGTM

File: docs/legacy/getting-started-apm-server.asciidoc
Comment (bmorelli25) [reply docs/legacy/getting-started-apm-server.asciidoc:239]: 👍 Made more prominent and will add (more prominently) to release notes as well.

**Comment by bmorelli25** [comment]
This PR is finally unblocked and ready for review.

**Review by axw — APPROVED** [review]
:man_cook: 

Just a couple of minor comments from me.

File: changelogs/head.asciidoc
Comment (bmorelli25) [reply changelogs/head.asciidoc:7]: Ah. Yes, this is a good distinction to make.

**Pull request merged**
